xcode/Subconscious/Shared/Models/FormField.swift

```diff
@@ -38,6 +40,7 @@ struct FormField<Input: Equatable, Output>: ModelProtocol {
     var defaultValue: Input
     /// Should be a pure, static function
     var validate: FormFieldValidator<Input, Output>
+    var isValid: Bool = false
```

Comment:
By making this a field, instead of derived, we can mark a field as invalid via a store action.

This is used to present the "Could not redeem invite code" error on a field.

Reply:
Making it an actual data field also allows us to trigger a re-render on change of validation.

Reply:
One question this raises for future is if it is worth having the validation lambda be part of the struct.

As an alternative, the parent update function could do whatever validation is needed and manually toggle the `isValid` field of the form field. This would move the business logic of validation out of the form field component, and into the specific parent view component. May allow us to side-step the generic in the form field model.

Reply:
Agreed, I have been deferring a change like this until we inevitably need validation that can produce different messages (instead of binary true/false) but I don't like carrying the closure around like this. 

We might also be able to leverage a protocol to make validation cleaner?

Reply:
Added an issue to track this https://github.com/subconsciousnetwork/subconscious/issues/779